Content.Server/Implants/SubdermalImplantSystem.cs

```diff
@@ -107,42 +111,84 @@ private void OnScramImplant(EntityUid uid, SubdermalImplantComponent component,
-            targetCoords = entityCoords.Offset(_random.NextAngle().ToVec() * distance);
+            _xform.SetWorldPosition(ent, targetCoords.Value.Position);
+            _xform.AttachToGridOrMap(ent, xform);
+            _audio.PlayPvs(implant.TeleportSound, ent);
+            args.Handled = true;
+        }
+    }
 
-            // prefer teleporting to grids
-            if (!_mapManager.TryFindGridAt(targetCoords, out var gridUid, out var grid))
-                continue;
+    private MapCoordinates? SelectRandomTileInRange(TransformComponent userXform, float radius)
+    {
+        var userCoords = userXform.Coordinates.ToMap(EntityManager, _xform);
+        var grids = _lookupSystem.GetEntitiesInRange<MapGridComponent>(userCoords, radius).ToList();
+        _random.Shuffle(grids);
 
-            // the implant user probably does not want to be in your walls
-            var valid = true;
-            foreach (var entity in grid.GetAnchoredEntities(targetCoords))
+        // Give preference to the grid the entity is currently on.
+        var idx = grids.FindIndex(grid => grid.Owner == userXform.GridUid);
+        if (idx != -1)
+        {
+            if (_random.Prob(0.66f))
+            {
+                (grids[0], grids[idx]) = (grids[idx], grids[0]);
+            }
+            else
             {
-                if (!_physicsQuery.TryGetComponent(entity, out var body))
-                    continue;
+                (grids[^1], grids[idx]) = (grids[idx], grids[^1]);
+            }
+        }
 
-                if (body.BodyType != BodyType.Static ||
-                    !body.Hard ||
-                    (body.CollisionLayer & (int) CollisionGroup.Impassable) == 0)
-                    continue;
+        MapCoordinates? targetCoords = null;
 
-                valid = false;
-                break;
+        foreach (var grid in grids)
+        {
+            var valid = false;
+
+            var range = (float) Math.Sqrt(radius);
+            var box = Box2.CenteredAround(userCoords.Position, new Vector2(range, range));
+            var tilesInRange = _mapSystem.GetTilesEnumerator(grid.Owner, grid.Comp, box, false);
+            var tileList = new List<TileRef>();
+
+            while (tilesInRange.MoveNext(out var tile))
+            {
+                tileList.Add(tile);
             }
+
+            _random.Shuffle(tileList);
+
+            foreach (var tile in tileList)
+            {
+                valid = true;
+                foreach(var entity in _mapSystem.GetAnchoredEntities(grid.Owner, grid.Comp, tile.GridIndices))
+                {
+                    if (!_physicsQuery.TryGetComponent(entity, out var body))
+                        continue;
+
+                    if (body.BodyType != BodyType.Static ||
+                        !body.Hard ||
+                        (body.CollisionLayer & (int) CollisionGroup.Impassable) == 0)
+                        continue;
+
```

Comment:
Is this supposed to not intersect with anything blocking the entity or just these as it's still possible to warp into tables or closets for example and get stuck.

Reply:
For this I just took the check that was there behavior. I hadn't heard any complaints about it warping you on stuff like tables so I just left it and assumed it just wanted to stop people from going into walls like the original comment there said. I'll change it.